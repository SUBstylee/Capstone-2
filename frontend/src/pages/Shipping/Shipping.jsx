import { useState } from "react";
import { useNavigate } from "react-router-dom";
import {Form,Button} from 'react-bootstrap';
import { useDispatch,useSelector } from "react-redux";
import { saveShippingAddress } from "../../actions/cartActions";
import FormContainer from "../../components/FormContainer/FormContainer";
import CheckoutSteps from "../../components/CheckoutSteps/CheckoutSteps";
import MetaWrapper from "../../components/MetaWrapper/MetaWrapper";
import CustomInput from "../../components/CustomInput/CustomInput";

const Shipping = () => {
    const cart=useSelector(state=>state.cart);
    const {shippingAddress}=cart;
    const dispatch=useDispatch();
    const navigate=useNavigate();

    const [address,setAddress]=useState(shippingAddress.address||'');
    const [city,setCity]=useState(shippingAddress.city||'');
    const [postalCode,setPostalCode]=useState(shippingAddress.postalCode||'');
    const [country,setCountry]=useState(shippingAddress.country||'');

    const submitHandler=(e)=>{
        e.preventDefault();
        dispatch(saveShippingAddress({address,city,postalCode,country}));
        navigate('/payment');
    };

    return (
        <div className="Shipping">
            <MetaWrapper title='TAA-Shipping'/>
            <FormContainer>
                <CheckoutSteps step1 step2/>
                <h1>Shipping</h1>
                <Form onSubmit={submitHandler}>
                    <CustomInput 
                        id='address' 
                        type='text' 
                        autoComplete="address" 
                        name='address' 
                        labelText='Address' 
                        value={address} 
                        required 
                        onChange={(e)=>setAddress(e.target.value)}/>
                    <CustomInput 
                        id='city' 
                        type='text' 
                        autoComplete="home city" 
                        name='city' 
                        labelText='City' 
                        value={city} 
                        required 
                        onChange={(e)=>setCity(e.target.value)}/>
                    <CustomInput 
                        id='postalCode' 
                        type='number' 
                        autoComplete="postal-code" 
                        name='postalCode' 
                        labelText='Postal Code' 
                        value={postalCode} 
                        required 
                        onChange={(e)=>setPostalCode(e.target.value)}/>
                    <CustomInput 
                        id='country' 
                        type='text' 
                        autoComplete="country" 
                        name='country' 
                        labelText='Country' 
                        value={country} 
                        required 
                        onChange={(e)=>setCountry(e.target.value)}/>
                    <Button className='mt-2' type='submit' variant="primary">Continue</Button>
                </Form>
            </FormContainer>   
        </div>
    );
};

export default Shipping