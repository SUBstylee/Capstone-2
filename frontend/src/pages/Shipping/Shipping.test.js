import { render } from '@testing-library/react';
import Shipping from './Shipping.jsx';
import store from '../../store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

test("renders ProductDetails component without crashing", () => {
    const { getByText } = render(
        <Provider store={store}><BrowserRouter><Shipping /></BrowserRouter></Provider>
    );

    expect(getByText("Continue")).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <Provider store={store}><BrowserRouter><Shipping /></BrowserRouter></Provider>
    );
    expect(asFragment()).toMatchSnapshot();
});