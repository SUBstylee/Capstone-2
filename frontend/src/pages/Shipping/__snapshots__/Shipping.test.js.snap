// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Shipping"
  >
    <div
      class="FormContainer"
    >
      <div
        class="container"
      >
        <div
          class="justify-content-md-center row"
        >
          <div
            class="col-md-6 col-12"
          >
            <div
              class="CheckoutSteps"
            >
              <div
                class="justify-content-center mb-4 nav"
              >
                <div
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/login"
                    href="/login"
                  >
                    Sign In
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    class="nav-link"
                    data-rr-ui-event-key="/shipping"
                    href="/shipping"
                  >
                    Shipping
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    role="button"
                  >
                    Payment
                  </a>
                </div>
                <div
                  class="nav-item"
                >
                  <a
                    aria-disabled="true"
                    class="nav-link disabled"
                    role="button"
                  >
                    Place Order
                  </a>
                </div>
              </div>
            </div>
            <h1>
              Shipping
            </h1>
            <form
              class=""
            >
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="address"
                  id="address"
                  name="address"
                  required=""
                  type="text"
                  value=""
                />
                <label
                  for="address"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    A
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    d
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    d
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    r
                  </span>
                  <span
                    style="transition-delay: 200ms;"
                  >
                    e
                  </span>
                  <span
                    style="transition-delay: 250ms;"
                  >
                    s
                  </span>
                  <span
                    style="transition-delay: 300ms;"
                  >
                    s
                  </span>
                </label>
              </div>
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="home city"
                  id="city"
                  name="city"
                  required=""
                  type="text"
                  value=""
                />
                <label
                  for="city"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    C
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    i
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    t
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    y
                  </span>
                </label>
              </div>
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="postal-code"
                  id="postalCode"
                  name="postalCode"
                  required=""
                  type="number"
                  value=""
                />
                <label
                  for="postalCode"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    P
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    o
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    s
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    t
                  </span>
                  <span
                    style="transition-delay: 200ms;"
                  >
                    a
                  </span>
                  <span
                    style="transition-delay: 250ms;"
                  >
                    l
                  </span>
                  <span
                    style="transition-delay: 300ms;"
                  >
                     
                  </span>
                  <span
                    style="transition-delay: 350ms;"
                  >
                    C
                  </span>
                  <span
                    style="transition-delay: 400ms;"
                  >
                    o
                  </span>
                  <span
                    style="transition-delay: 450ms;"
                  >
                    d
                  </span>
                  <span
                    style="transition-delay: 500ms;"
                  >
                    e
                  </span>
                </label>
              </div>
              <div
                class="CustomInput"
              >
                <input
                  autocomplete="country"
                  id="country"
                  name="country"
                  required=""
                  type="text"
                  value=""
                />
                <label
                  for="country"
                >
                  <span
                    style="transition-delay: 0ms;"
                  >
                    C
                  </span>
                  <span
                    style="transition-delay: 50ms;"
                  >
                    o
                  </span>
                  <span
                    style="transition-delay: 100ms;"
                  >
                    u
                  </span>
                  <span
                    style="transition-delay: 150ms;"
                  >
                    n
                  </span>
                  <span
                    style="transition-delay: 200ms;"
                  >
                    t
                  </span>
                  <span
                    style="transition-delay: 250ms;"
                  >
                    r
                  </span>
                  <span
                    style="transition-delay: 300ms;"
                  >
                    y
                  </span>
                </label>
              </div>
              <button
                class="mt-2 btn btn-primary"
                type="submit"
              >
                Continue
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
